### ISSUE TITLE
CDH spark versions don't compile

### ISSUE THREAD

**Issue by agibsonccc** [andypetrella/spark-notebook#655]
Seeing this:
https://gist.github.com/agibsonccc/b5e3d9b101982ce297b7ce548002847a

Built with:
sbt -Dscala.version=2.10.5 -Dspark.version=1.6.0-cdh5.7.1 -Dhadoop.version=2.6.0-cdh5.7.1 clean dist

**Comment by huitseeker** [comment]
This one is a consequence of 1.6.1 rather than CDH:
https://github.com/apache/spark/commit/4a46b8859d3314b5b45a67cdc5c81fecb6e9e78c
https://issues.apache.org/jira/browse/SPARK-11563

**Comment by andypetrella** [comment]
@huitseeker we need to change something?

**Comment by huitseeker** [comment]
Yup, class Server is gone : https://github.com/apache/spark/commit/4a46b8859d3314b5b45a67cdc5c81fecb6e9e78c#diff-15ebfc86607787462983cf29af227de5L116

**Comment by andypetrella** [comment]
@huitseeker so ... after 1.6.1? I didn't check... if so it's rough in the middle of a minor :-)

**Comment by huitseeker** [comment]
Oh, no, the fix is in 2.0.0. But that may actually mean more that we should update the pointer to 'spark-last/' used in the error encountered above : this compilation path should point to a 1.6.x specific repo.

**Comment by andypetrella** [comment]
@huitseeker I guess that cloudera back ported this change in their 1.6.1 ... very clever -_-, and I guess relevant plus helpful :-S

**Comment by vidma** [comment]
I don't think we should worry much about CDH spark.

we use CDH hadoop, but not CDH-spark because it used to be crap, and always horribly late.

compiling spark is not hard at all! and you can choose proper options.

```bash
export MAVEN_OPTS="-Xmx2g -XX:MaxPermSize=512M -XX:ReservedCodeCacheSize=512m"

#  ( for some mvn versions `./make-distribution.sh` is slightly buggy, so last gzipping stage fails, but everything work, you can gz yourself)
./make-distribution.sh --name spark-1.6.1 --tgz -Pyarn -Phadoop-2.6 -Dhadoop.version=2.6.0-cdh5.7.0 -Phive -DskipTests clean package

# or same without make distribution script
build/mvn -Pyarn -Phadoop-2.6 -Dhadoop.version=2.6.0-cdh5.7.0 -Phive -DskipTests clean package
```

**Comment by huitseeker** [comment]
Pinging @srowen just in case he has some insight as to the backport (Hi Sean, commit SPARK-11563, PR, are referenced above and you can just ping me for details) .

**Comment by srowen** [comment]
Here is where you can see all the commits in the CDH 5.7 Spark (1.6) branch that are on top of the upstream rebase: http://archive.cloudera.com/cdh5/cdh/5/spark-1.6.0-cdh5.7.0.CHANGES.txt
SPARK-11563 is in there. I think Marcelo pulled it in to fix some other problems before 2.x. It does remove some internal classes, but I think the idea is that they are not an API.

@vidma CDH Spark is always the latest available when the minor release goes out and the release cadence roughly matches Spark's (3-4 months). I don't know what problems you're seeing (except this), but I don't think this is a problem. The audience for the bleeding edge can always build their own.

I think you'll have to adjust to this change for 2.x in some fashion anyway.

**Comment by andypetrella** [comment]
@srowen you're right! I just wanted to be sure you backported this.
That's cool by me (I've started looking at the necessary changes for 2.0...)

thanks man
